product_supplierinfo_for_customer_sale/models/product_product.py

```diff
@@ -13,14 +13,15 @@ class ProductProduct(models.Model):
     def name_search(self, name='', args=None, operator='ilike', limit=80):
         res = super(ProductProduct, self).name_search(
             name, args=args, operator=operator, limit=limit)
-        if res or not self._context.get('partner_id'):
+        if not name or not self._context.get('partner_id'):
             return res
         partner_id = self._context['partner_id']
         supplierinfo = self.env['product.supplierinfo'].search(
             [('name', '=', partner_id), '|', ('product_code', operator, name),
              ('product_name', operator, name)])
         product = self.search(
-            [('product_tmpl_id', '=', supplierinfo.product_tmpl_id.id)],
+            [('product_tmpl_id', 'in',
+              supplierinfo.mapped('product_tmpl_id').ids)],
             limit=limit)
-        res = product.name_get()
-        return res
+        res.extend(product.name_get())
+        return list(set(res))
```

Comment:
Sorry, just one more change... 
in order to improvement the performance here we should use this transformation just if is really need it.
So IMHO is better use:
```python
if products:
    res.extend(products.name_get())
    res = list(set(res))
return res
```


We can save a lot of seconds by method executed, you could calculate it with:
```bash
python -m timeit -s "res=[(1, 'A'),(2, 'B'),(3, 'C'),(2, 'B'),]" "res=list(set(res))"
1000000 loops, best of 3: 0.596 usec per loop
```